hazelcast-client/src/main/java/com/hazelcast/client/connection/AddressProvider.java

```diff
@@ -25,8 +26,8 @@
 public interface AddressProvider {
 
     /**
-     * @return Collection of InetSocketAddress
+     * @return The possible member addresses to connect to.
      */
-    Collection<InetSocketAddress> loadAddresses();
+    Collection<Address> loadAddresses();
```

Comment:
Is it ok to change a public interface in a patch release?

Reply:
> **Compile time compatibility:** patch, minor
> That means we should not break compile time compatibility in patch and minor releases.
> **Run time (binary) compatibility:** patch
> That means we should not break run-time compatibility in patch releases.

I don't think it's allowed. Is this even compile time compatible? It may at least not be runtime compatible (replace 3.8.5 JAR with 3.8.6, no re-compile).

Reply:
It is actually only internally used api and it is not really a public API that the user can use.

Reply:
Then we should annotate this as `@PrivateApi`